bin/ci/run-tests.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -u
+
+# Don't attempt to build the following plans. They have resource requirements
+# or build times that exceed the currently available resources on our CI
+# infrastructure.
+PLAN_BLACKLIST=(
+ glibc
+ gcc
+ ghc
+ ghc710
+ ghc710-bootstrap
+ ghc80
+ ghc82
+ ghc82-bootstrap
+ ghc84
+ ghc86
+ kubernetes
+ mongodb
+ mysql
+)
+plan="$(basename "$1")"
+
+HAB_ORIGIN=ci
+export HAB_ORIGIN
+
+for bl in "${PLAN_BLACKLIST[@]}"; do
+  if [[ "${plan}" == "${bl}" ]]; then
+    echo "--- :bangbang: [$plan] Skipping build"
+    # If we're running in buildkite, we also want to annotate the build
+    if [[ "${CI:-}" == "true" ]]; then
+      buildkite-agent annotate --style 'warning' ":bangbang: ${plan} found in build blacklist. Skipping build. "
+    fi
+    exit 0;
+  fi
+done
+
+if [ ! -f "$plan/tests/test.sh" ]; then
+  buildkite-agent annotate --style 'warning' ":warning: :linux: ${plan} has no Linux tests to run. This is currently permitted but please consider adding some tests."
```

Comment:
Oops, this is missing an `exit 0` to exit harmlessly.
I deliberately intended to have this skip the build - because the build is only here to have an artifact for the tests to run against. If there's no tests, the build is pointless. Unless we're counting it as an extra implicit test. But I would prefer that the test cases capture 100% of the testing scope.

Reply:
That was my original thought as well, but after pondering for a bit I think my preference would be to have it always build and attempt to run tests if they exist. 

a) We verify the change builds. We can't rely on the `DO_CHECK` CI builds to provide that verification, as those should "soft fail" if the do_check fails and we don't have a way of separating build failures from check failures. 

b) it isn't an entirely implicit test,  we expect the software to build in order to test it (restatement of a). The way I think of it is:
# Build "foo". I expect the build to be successful
# Test "origin/foo/version/release".  No tests defined. 0 of 0 tests passed, success.

Over time, we'd want to change 0 tests defined to a failure case, but not until we have sufficient coverage. We'd also want to make it super simple for a contributor to just drop in appropriate tests. We're making huge strides, but we're not there yet 😸

Reply:
Added a commit to make this task `build-and-test` instead of just `test` so it's called out that the build is a required part of this check.
Also made the lack of tests a soft fail.